CMakeLists.txt

```diff
@@ -34,10 +34,7 @@ find_package(Boost REQUIRED COMPONENTS iostreams)
 find_package(Ceres REQUIRED COMPONENTS SuiteSparse)
 find_package(Eigen3 REQUIRED)
 find_package(LuaGoogle REQUIRED)
-# find_package(Protobuf 3.0.0 REQUIRED)
-set(PROTOBUF_PROTOC_EXECUTABLE /opt/protobuf3/bin/protoc)
-set(PROTOBUF_INCLUDE_DIR /opt/protobuf3/include)
-set(PROTOBUF_LIBRARY /opt/protobuf3/lib/libprotobuf.so)
+find_package(Protobuf 3.0.0 REQUIRED)
```

Comment:
Is this the correct version for cmake? I think we rely on Protobuf 3.4, but unsure if the protobuf cmake supports this version.

Reply:
It seems like Cmake's FindProtobuf parses a header file (based on discovered/specified location of Protobuf) to find the version string. The version here is from the upstream repository and just means that protobuf3 is required. We could require a specific/more up to date version of protobuf here but I don't know if we have to or need to since we specify the protobuf library at build time.

Reply:
AFAIK protobuf 3.4 is explicitly required, but @AlexReimann should have a more informed opinion on that.

Of course we can always fail during build time, but if we can fail earlier, that should still be preferred.

Reply:
Comes from here: https://google-cartographer.readthedocs.io/en/latest/#system-requirements

Reply:
Yes, I just restored the original line. Even the upstream master still uses 3.0.0.